server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1474,8 +1474,16 @@ private DateMathExpressionResolver() {
 
         public static List<String> resolve(final Context context, List<String> expressions) {
             List<String> result = new ArrayList<>(expressions.size());
+            boolean wildcardSeen = false;
             for (String expression : expressions) {
-                result.add(resolveExpression(expression, context::getStartTime));
+                if (expression.charAt(0) == '-' && wildcardSeen) {
```

Comment:
Same comment about the modified code, can the expression ever be an empty string? I'm assuming we'll error out earlier for something like `-<index-{now/d}>,,*`.

Reply:
This is a very good point.
Previously the code in the `WildcardExpressionResolver` had a check for this: https://github.com/elastic/elasticsearch/blob/d1d0f6d6230df3fdb47edafdab1c4964693e3e98/server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L1202 .
But now, since the datemath has to check for exclusions in the same way, and that datemath eval actually runs before wildcard eval, I copied the empty name check here from the wildcard resolver, see [48c3ac4](https://github.com/elastic/elasticsearch/pull/90298/commits/48c3ac4a21d618063dd7a9c88c18e7d1638f36ab) . I don't think this can be hit in practice, but it is necessary to maintain the same "interface".

Reply:
Turns out [48c3ac4](https://github.com/elastic/elasticsearch/pull/90298/commits/48c3ac4a21d618063dd7a9c88c18e7d1638f36ab) is wrong. It does not not cater to the `_all` pattern...
I reverted that and pushed [469aa8f](https://github.com/elastic/elasticsearch/pull/90298/commits/469aa8fac5a9831dc92c14b51771b4694a4af432) which is a much simpler guard for empty.